test/micro/org/openjdk/bench/java/lang/classfile/Utf8EntryWriteTo.java

```diff
@@ -0,0 +1,122 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.bench.java.lang.classfile;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.lang.classfile.constantpool.ConstantPoolBuilder;
+import java.lang.classfile.*;
+import java.lang.constant.*;
+import java.nio.charset.StandardCharsets;
+import java.util.HexFormat;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import static java.lang.classfile.ClassFile.*;
+import static java.lang.constant.ConstantDescs.*;
+
+import jdk.internal.classfile.impl.*;
+/**
+ * Test various operations on
+ */
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Warmup(iterations = 1, time = 2)
+@Measurement(iterations = 3, time = 1)
+@Fork(jvmArgsAppend = "--enable-preview", value = 3)
+@State(Scope.Thread)
+public class Utf8EntryWriteTo {
+    static final ClassDesc STRING_BUILDER = ClassDesc.ofDescriptor("Ljava/lang/StringBuilder;");
+    static final MethodTypeDesc MTD_append = MethodTypeDesc.of(STRING_BUILDER, CD_String);
+    static final MethodTypeDesc MTD_String = MethodTypeDesc.of(CD_String);
+    static final ClassDesc CLASS_DESC = ClassDesc.ofDescriptor("Lorg/openjdk/bench/java/lang/classfile/String$$StringConcat;");
+
+    @Param({"ascii", "utf8_2_bytes", "utf8_3_bytes", "emoji"})
+    public String charType;
+    public String[] constants;
+
+    @Setup
+    public void setup() throws Exception {
+        byte[] bytes = HexFormat.of().parseHex(
+                switch (charType) {
+                    case "ascii"        -> "78";
+                    case "utf8_2_bytes" -> "c2a9";
+                    case "utf8_3_bytes" -> "e6b8a9";
+                    case "emoji"        -> "e29da3efb88f";
+                    default -> throw new IllegalArgumentException("bad charType: " + charType);
+                }
+        );
+        String s = new String(bytes, 0, bytes.length, StandardCharsets.UTF_8);
+        constants = new String[128];
+        for (int i = 0; i < constants.length; i++) {
+            constants[i] = s.repeat(32);
+        }
```

Comment:
I think this micro is testing too much irrelevant parts of classfile generation. I would recommend these changes:
1. Compute a `ConstantPoolBuilder` in `setup`, like:
```java
poolBuilder = ConstantPoolBuilder.of();
thisClass = poolBuilder.classEntry(CLASS_DESC);
for (var s : constants) {
    poolBuilder.utf8Entry(s);
}
```
2. In benchmark, test like:
```java
@Benchmark
public byte[] writeTo() {
    return cf.build(thisClass, poolBuilder, clb -> {});
}
```
This should eradicate the cost around method or instruction generation while still ensuring the constants are written to the class file as part of the constant pool.